### PR TITLE
7-new-notifications

### PR DESCRIPTION
**Pull request by zatziky**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ try{
  REST | js | Description  
 ----------------------------|---|-------------
 POST&nbsp;/v1/messages | `messages.send(message)` | Send a message to a contact. There are different message types for every platform ([FB](https://docs.amio.io/v1.0/reference#facebook-messenger-messages-text), [Viber](https://docs.amio.io/v1.0/reference#viber-messages-text)).
+POST&nbsp;/v1/notifications | `notifications.send(notification)` | Send a notification to a contact. There are notifications for [FB](https://docs.amio.io/v1.0/reference#facebook-messenger-notifications-send-notification) only.
```
Comment (kuceram) [inline README.md]: I would not mention for which platforms it is eligible. User can see it in docs. I would rather lead them to general notification link: https://docs.amio.io/v1.0/reference#notifications-send-notification

File: lib/api/message.js [hunk 2]
```diff
@@ -1,22 +1,18 @@
-class Message {
+const Request = require('./request')
+
+class Message extends Request {
 
   constructor(httpClient){
+    super()
```
Comment (kuceram) [inline lib/api/message.js]: `httpClient` can be assigned in `Request` class.

File: lib/api/notification.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const Request = require('./request')
+
+class Notification extends Request{
+
+  constructor(httpClient){
+    super()
+    this.httpClient = httpClient
+  }
+
+  send(notification, accessToken = '') {
```
Comment (kuceram) [inline lib/api/notification.js]: Default value of `accessToken` should be null.

File: lib/api/message.js [hunk 4]
```diff
@@ -1,22 +1,18 @@
...
     this.httpClient = httpClient
   }
 
   send(message, accessToken = '') {
-    const headers = createHeaders(accessToken)
```
Comment (kuceram) [inline lib/api/message.js]: Default value of `accessToken` should be null.

File: lib/api/notification.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const Request = require('./request')
+
+class Notification extends Request{
```
Comment (kuceram) [inline lib/api/notification.js]: wrong formatting, missing space

Comment (kuceram) [inline lib/api/notification.js]: Should you define notification types to be used by sdk user?

File: lib/api/request.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class Request {
+
+  createHeaders(accessToken) {
```
Comment (kuceram) [inline lib/api/request.js]: I prefer `buildHeaders` here. But it's up to you...

File: lib/connector.js [hunk 7]
```diff
@@ -1,14 +1,16 @@
 const amioHttpClient = require('./http/amio.http-client')
 const Message = require('./api/message')
+const Notification = require('./api/notification')
 
 class AmioConnector {
 
   constructor (opts = {}) {
     const accessToken = opts.accessToken || null
-    const baseUrl = opts.baseUrl
+    const baseUrl = opts.baseUrl || 'https://api.amio.io' // TODO find out if we have cases when we need to override baseUrl
```
Comment (kuceram) [inline lib/connector.js]: Remove TODO...

File: test/config.js [hunk 8]
```diff
@@ -2,7 +2,6 @@ const forEachObjIndexed = require('ramda/src/forEachObjIndexed')
 
 // we set config for amio+sdk_tests@amio.io in prod
 const config = {
-  CONNECTOR_API_BASE_URL: 'https://api.amio.io',
```
Comment (kuceram) [inline test/config.js]: I would not mind to have it here rather then hardcoded somewhere in code..

File: test/e2e/connector.test.js [hunk 9]
```diff
@@ -37,14 +40,26 @@ describe('Amio API Connector', function () {
     })
   })
 
+  describe('notifications', () => {
+    it('send a notification', async () => {
+
+      const type = 'typing_on'
```
Comment (kuceram) [inline test/e2e/connector.test.js]: what about to test all notification types?

File: lib/api/notification.js (hunk 5 above)
Comment (zatziky) [reply lib/api/notification.js]: I would leave this part on developers that will use Amio. Let's leave it generic untill we collect more info about how our users use this lib. ;)

File: lib/api/request.js (hunk 6 above)
Comment (zatziky) [reply lib/api/request.js]: je to factory metoda, tak to nechám tak jak je. `build` by se hodil pro builder pattern.

File: test/config.js (hunk 8 above)
Comment (zatziky) [reply test/config.js]: It is just a config for tests. I will keep it where it is.

File: test/e2e/connector.test.js (hunk 9 above)
Comment (zatziky) [reply test/e2e/connector.test.js]: I am just testing that the request for a notification passes. All notifTypes are tested in amio-communicator.

File: lib/api/request.js (hunk 6 above)
Comment (kuceram) [reply lib/api/request.js]: Wie bitte? :-D

**Comment by kuceram** [comment]
I don't see any changes and it is already merged. :-/

File: lib/api/request.js (hunk 6 above)
Comment (zatziky) [reply lib/api/request.js]: Oh shit, it's public! we shouldn't use bad words and Czech! :-O 

I said: "It's a factory method. `build` is usually used in combination with the builder pattern".

**Pull request merged**